src/services/paramValidation.ts

```diff
@@ -30,10 +30,12 @@ function getPathParams(path: Path, url: string): Record<string, unknown> {
   const decodedValueFromUrl = decodeURIComponent(url)
 
   for (const [key, param] of Object.entries(path.params)) {
+    const isOptional = key.startsWith('?')
+    const paramName = isOptional ? key.slice(1) : key
     const stringValue = getParamValueFromUrl(decodedValueFromUrl, path.toString(), key)
-    const formattedValues = getParamValue(stringValue, param)
+    const formattedValues = getParamValue(stringValue, param, isOptional)
```

Comment:
previously the `param` argument was modified to include context of if value is optional. Now that logic exists outside of the `param`, so I added an optional 3rd argument to `getParamValue` and `setParamValue`.